### ISSUE TITLE
evmc_execute should take evmc_tx_context as an parameter

### ISSUE THREAD

**Issue by axic** [ethereum/evmc#120]
I think all those execution details should be available to the client prior to execution. Do you think any client can be savings by assuming the code will not need those and only lazily populate `evmc_tx_context` on the first call to to `get_tx_context`?

**Comment by chfast** [comment]
This comes down to https://github.com/ethereum/evmjit/issues/110.

**Comment by axic** [comment]
Potentially this may want to be benchmarked and as such would be postponed to ABIv7.

**Comment by chfast** [comment]
I believe it is good change. It saves us trouble caching this on VM side.
Also, this can be handled very efficiently on the Host side - it requires to create the `tx_context` once per block, and then update the 2 fields than depend on transaction. If you process transactions in parallel, you need multiple copies per execution thread.
